File: app/src/main/java/com/ivanmagda/habito/barchart/formatters/YearAxisValueFormatter.java
package com.ivanmagda.habito.barchart.formatters;

import com.github.mikephil.charting.components.AxisBase;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

import static com.ivanmagda.habito.utils.HabitoDateUtils.getCurrentCalendar;

public final class YearAxisValueFormatter extends HabitoBaseIAxisValueFormatter {

    private static SimpleDateFormat FORMATTER = new SimpleDateFormat("MMM", Locale.getDefault());

    @Override
    public String getFormattedValue(float value, AxisBase axis) {
        String yearString = FORMATTER.format(new Date(getDateForValue(value)));
        return yearString.substring(0, 1);
    }

    @Override
    public long getDateForValue(float value) {
        Calendar calendar = getCurrentCalendar();
        calendar.set(Calendar.MONTH, (int) value);
        return calendar.getTimeInMillis();
    }

}

File: app/src/main/java/com/ivanmagda/habito/barchart/formatters/YearAxisValueFormatter.kt
package com.ivanmagda.habito.barchart.formatters

import com.github.mikephil.charting.components.AxisBase
import com.ivanmagda.habito.util.HabitoDateUtils
import java.text.SimpleDateFormat
import java.util.*

class YearAxisValueFormatter : HabitoBaseIAxisValueFormatter() {

    override fun getFormattedValue(value: Float, axis: AxisBase): String {
        val yearString = FORMATTER.format(Date(getDateForValue(value)))
        return yearString.substring(0, 1)
    }

    override fun getDateForValue(value: Float): Long {
        val calendar = HabitoDateUtils.currentCalendar
        calendar.set(Calendar.MONTH, value.toInt())

        return calendar.timeInMillis
    }

    companion object {
        private val FORMATTER = SimpleDateFormat("MMM", Locale.getDefault())
    }
}
